### ISSUE TITLE
Send a notification to the author if a PR has CI failures

### ISSUE THREAD

**Issue by wilzbach** [dlang-bots/dlang-bot#123]
CircleCi already does this and it's very handy as one doesn't need to observe the CI status.
Of course we would need to exclude CodeCov and make sure that we only send the notification if its a real failure (i.e. fix our transient CI problem).

**Comment by MartinNowak** [comment]
Do we really want to spam people without explicit permission? It also has the risk to ruin notifications as one of the few communication mediums.

**Comment by wilzbach** [comment]
We currently label PRs with have two or more build failures as "needs work" as I wasn't so sure about spamming people either. Maybe we could do it once a month or when a rebase is needed?

**Comment by CyberShadow** [comment]
I've been in the following situation:

- Run into a problem that's solvable by a patch in a D component
- Write a patch against the D component and submit it as a PR
- Go on with whatever I was doing

Meanwhile, the CI fails because I forgot some silly thing.

The situation:
- D maintainers are waiting for me to notice that the CI is failing and fix the PR.
- I am waiting for a ping or some kind of comment from a D maintainer that I need to do something with the PR (or that it has been approved/merged).

Maybe in addition to the immediate welcome comment that dlang-bot posts, we can aggregate more useful information that is collected after a CI runs, which would include
- CI status (and maybe some analysis of CI failures)
- Aggregated coverage (if CI succeeds)
- Pinging reviewers (if we decide we want some system on top of CODEOWNERS)

?

**Comment by Geod24** [comment]
Personally, I routinely go to https://github.com/pulls to get a list of things I have "on hold". It shows the CI status so if it's red, I know I have to take action. I'd rather leave it at that than spam people, as Martin mentioned.

**Comment by CyberShadow** [comment]
Yes, I guess that would work for people who do all their work on GitHub, but I think that's not everyone.
